server/src/ability/ability.util.ts

```diff
@@ -59,6 +59,17 @@ const simpleAbilityCheck: OperationAbilityChecker = async (
   prisma,
   data,
 ) => {
+  // TODO: Replace user by workspaceMember and remove this check
+  if (
+    modelName === 'User' ||
+    modelName === 'UserSettings' ||
+    modelName === 'Workspace' ||
+    typeof data === 'boolean'
+  ) {
+    return true;
+  }
+
+  // We need to consider optional relations because the data will be a bool, `true` of `false`
```

Comment:
On close examination of this function, all the other logic is redundant if the model is `User`, `UserSettings` or `Workspace`. If there are items found, it will only iterate over the first item and return `true`. On the other hand, if there are no items found, it will just return `true`, so why not just return true at the top?

If the data passed is a boolean e.g in the case where we're disconnecting an optional property
```json
{
  "company": {
    "disconnect": true
  }
}
```

The data value of `simpleAbilityCheck` will be true which will in term make the `whereOption` invalid
```ts
const items = await prisma[entity as string].findMany({
    where: {
      OR: normalizedOperations, // the problem is here, OR: [ true | false ]
    },
  });
```

![optimizations-1](https://github.com/twentyhq/twenty/assets/140154534/5214f6f7-5760-4523-9671-69db33dd3fc8)

Reply:
Ok, looks legit, please separate:
```
 if (
    modelName === 'User' ||
    modelName === 'UserSettings' ||
    modelName === 'Workspace' ||
  ) {
    return true;
  }
```

from
```
 if (typeof data === 'boolean')
  {
    return true;
  }
```

and remove comment `// the problem is here, OR: [ true | false ]`